src/sbd-cluster.c

```diff
@@ -67,21 +70,149 @@ sbd_plugin_membership_dispatch(cpg_handle_t handle,
 #endif
 
 #if SUPPORT_COROSYNC
+#include <corosync/cmap.h>
+
+static bool two_node = false;
+static bool ever_seen_both = false;
+static cmap_handle_t cmap_handle = 0;
+static cmap_track_handle_t track_handle = 0;
+static int cpg_membership_entries = -1;
+static GSource *cmap_source = NULL;
+
+void
+sbd_cpg_membership_health_update()
+{
+    if(cpg_membership_entries > 0) {
+        if (!two_node || !ever_seen_both || cpg_membership_entries > 1) {
```

Comment:
I'll would add a comment like that:
Caveat: This is not to replace the quorum-handling in pacemaker-servant.
It is just there to help out in the one and only case where 2Node-flag makes quorum coming from pengine via cib unusable for pacemaker-servant. Reason why it is not implemented there is that cluster-servant already interacts with corosync and pacemaker-servant just reads the cib and derives everything from there.
If !two_node or !ever_seen_both it falls back to the previous behavior.
ever_seen_both may be named badly but it is there to assure startup-cases and it is set once more than 1 node (most likely this node himself) has been seen.